File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.cpp
// Last updated: 7/16/2026, 4:39:17 PM
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        int i = 0;
        int j = 1;

        while (j<n){
            if(nums[i] != nums[j]){
                i++;
            nums[i] = nums[j];
            }
            j++;
        }
        return i + 1;
    }
};

File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.py
# Last updated: 8/26/2026, 10:03:59 AM
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l = []
        for i in nums:
            if not l or i != l[-1]:
                l.append(i)
        for i in range(len(l)):
            nums[i] = l[i]
        return len(l)
        